tests/fixtures_server.py

```diff
@@ -123,51 +123,69 @@ def get_patched_settings(temp_path: Path):
     return settings
 
 
+modules_to_patch = [
+    module[:-3].replace("/", ".").replace(".__init__", "")
+    for module in glob.glob(
+        os.path.join("fractal_server", "**/*.py"), recursive=True
+    )
+    if "migrations" not in module
+]
+
+# SETTINGS fixtures
+
+
 @pytest.fixture(scope="session", autouse=True)
-def override_settings(tmp777_session_path):
+async def set_default_test_settings(tmp777_session_path):
     tmp_path = tmp777_session_path("server_folder")
+    patched_settings = get_default_test_settings(tmp_path)
+    with pytest.MonkeyPatch.context() as mp:
+        for module in modules_to_patch:
+            mp.setattr(
+                f"{module}.get_settings",
+                lambda: patched_settings,
+                raising=False,
+            )
+        yield
 
-    settings = get_patched_settings(tmp_path)
 
-    def _get_settings():
-        return settings
+@pytest.fixture
+async def override_settings_startup(tmp777_session_path, monkeypatch, request):
+    if not request.__dict__.get("param"):
+        return
+
+    tmp_path = tmp777_session_path("server_folder")
+    patched_settings = get_default_test_settings(tmp_path)
 
-    Inject.override(get_settings, _get_settings)
-    try:
-        yield settings
-    finally:
-        Inject.pop(get_settings)
+    for k, v in request.param.items():
+        setattr(patched_settings, k, v)
```

Comment:
`pydantic` does not automatically perform checks upon `setattr` (unless we change its the model's Config block).
Maybe it's worth double-checking the model validity after this loop of updates, with an explicit call of the model init (or any other pydantic method for validation).